lib/influx.js

```diff
@@ -1,28 +1,31 @@
-const Influx = require('influx')
+const { InfluxDB, Point, HttpError } = require('@influxdata/influxdb-client')
+const { SetupAPI } = require('@influxdata/influxdb-client-apis')
+const { hostname } = require('node:os')
 const { Writable } = require('node:stream')
 
-let influx
+let influxWriteApi
+;(async () => {
+  influxWriteApi = await loadInfluxInstance()
+})()
+
 const stream = new Writable({
   async write(chunk, encoding, callback) {
     try {
-      await loadInfluxInstance()
-
       const currentWorkout = JSON.parse(chunk.toString())
 
-      const influxPoints = [currentWorkout].map((workout) => ({
-        measurement: process.env.APPLE_WATCH_INFLUX_MEASUREMENT,
-        tags: { type: workout.type, date: workout.startDate.substring(0, 10) },
-        fields: {
-          duration: workout.duration,
-          totalDistance: workout.totalDistance,
-          totalEnergyBurned: workout.totalEnergyBurned,
-          sourceName: workout.sourceName,
-          sourceVersion: workout.sourceVersion,
-        },
-        timestamp: new Date(workout.startDate),
-      }))
+      const influxPoints = [currentWorkout].map((workout) =>
+        new Point(process.env.APPLE_WATCH_INFLUX_MEASUREMENT)
+          .tag('type', 'date')
```

Comment:
I think the result is not what you expect:
```js
tags: { type: 'date' },
```

Use `tag` and give the name and value: 
```js
    .tag('type', workout.type)
    .tag('date', workout.startDate.substring(0, 10))
```
To get a result like this:
```js
tags: { type: 'cycling', date: '2024-10-16' },
```

Reply:
Thank you!